### ISSUE TITLE
Enhancing the Review Process

### ISSUE THREAD

**Issue by dylanfitz** [huboard/huboard-web#195]
I work with a team of about 10 people, and often we need someone to review an issue before it can get moved to "done". Currently, we just place it in review and "@" mention the person who needs to review that item, however, with a growing team that is managing much larger projects, this is becoming difficult to manage. 

We are looking for a way to add team members to an issue that are responsible for reviewing the work done in that issue. Some key information that would be helpful is if the reviewer has opened the issue since it was placed in review, and maybe an indicator that the user approves this work. It would also be helpful if these reviewers received a notification email when an issue is moved into "Review". It is important that the review is tied to a user, and not just a free floating feature. It is also important that more than one reviewer can be added to an issue.

**Comment by dahlbyk** [comment]
This is very similar to https://github.com/huboard/huboard/issues/612 - what do you think of https://github.com/huboard/huboard/issues/612#issuecomment-175121205?

**Comment by seanlinsley** [comment]
@dahlbyk I think it's important to assign the PR itself to the reviewer from inside Github instead of storing this information as metadata.

**Comment by dahlbyk** [comment]
> I think it's important to assign the PR itself to the reviewer from inside Github instead of storing this information as metadata.

I tend to agree, we're just limited to one GitHub-provided assignee slot. The idea would be to use metadata to support tracking a secondary set of assignees that HuBoard would let you swap into the GH Assignee field as appropriate for your team's process.

**Comment by seanlinsley** [comment]
What I had in mind was that from Huboard there could only be two assignees: the implementer and the reviewer. The implementer would continue to work as it currently does, but the reviewer would be stored as metadata until a PR was created, at which point the PR itself would be assigned to the reviewer.

My concerns with doing any more than that to fit various team processes are:
- the assignee(s) wouldn't be visible from Github
- the feature would take a lot longer to build, and maintain
- it would be harder for users to understand the feature, and configure it how they want

**Comment by dahlbyk** [comment]
> the reviewer would be stored as metadata until a PR was created, at which point the PR itself would be assigned to the reviewer.

Interesting...so you'd want to set the expected reviewer on the _issue_ so assignments can be automated on the PR. That hadn't occurred to me - I am intrigued. I'm not sure that we're completely set up to inspect PRs' referenced issues to track down a reviewer, but it's definitely an interesting idea.

As for "the assignee(s) wouldn't be visible from Github", I wonder if we could implement a DSL of sorts to track assignees "in the clear", e.g. something like this at the bottom of the issue (above our standard metadata comment):

> blah blah description we should fix this
>
> Assignees: @discorick @seanlinsley 
>

**Comment by seanlinsley** [comment]
Your idea to track assignees based on the text of the ticket sounds interesting, but it comes with technical challenges, e.g. it requires that Huboard listen to every modification of the ticket description and trigger changes based on that.

**Comment by dahlbyk** [comment]
How would you envision this working now that GitHub has added support for [multiple assignees](https://github.com/blog/2178-multiple-assignees-on-issues-and-pull-requests)?

**Comment by seanlinsley** [comment]
Cross-ref: https://github.com/huboard/huboard/issues/612

**Comment by seanlinsley** [comment]
It would be nice if Github had mentioned what their plans for the feature were. It's like they built the most basic version of the feature, without an idea of how it would actually fit into people's workflows.

I pinged them on Twitter to see if they have further plans: https://twitter.com/seanlinsley/status/736556701734699008

**Comment by dahlbyk** [comment]
> It would be nice if Github had mentioned what their plans for the feature were. It's like they built the most basic version of the feature, without an idea of how it would actually fit into people's workflows.

Perhaps we can sneak assignee metadata into the API before it's out of Preview? I wouldn't get my hopes up, but we can ask.

> I pinged them on Twitter to see if they have further plans: https://twitter.com/seanlinsley/status/736556701734699008

You might get more of a response from support@github.com.

**Comment by seanlinsley** [comment]
I just sent an email to Github support, asking that they respond on this ticket.

**Comment by seanlinsley** [comment]
Their response:

> Thanks for getting in touch. 

>I've logged your feedback for the team to review. 

>Being able to assign a primary assignee and/ or teams does seem like a useful idea.

>I can't promise if or when we would implement this but we always appreciate the feedback!
